src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -409,8 +409,10 @@ def calculate_deps(self) -> List[str]:
         # add an explicit dependency on the active manager. This will force to
         # re-deploy prometheus if the mgr has changed (due to a fail-over i.e).
         deps.append(self.mgr.get_active_mgr().name())
-        deps += [s for s in ['node-exporter', 'alertmanager', 'ingress']
+        deps += [s for s in ['node-exporter', 'alertmanager']
                  if self.mgr.cache.get_daemons_by_service(s)]
+        if len(self.mgr.cache.get_daemons_by_type('ingress')) > 0:
+            deps.append('ingress')
```

Comment:
ah subtle! the func names are unfortunate here:

- `get_daemons_by_service` is really `get_daemons_by_service_name`
- `get_daemons_by_type` is really `get_daemons_by_service_type` 

... the last one really seems to imply a param of a daemon type, when that's not actually the case.

Reply:
> 

upsss, good catch! I didn't know about that, I think we should definitely rename them as indeed they are very confusing